modules/oe_whitelabel_search/src/Plugin/Block/SearchBlock.php

```diff
@@ -0,0 +1,197 @@
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configFactory = $config_factory;
+    $this->formBuilder = $form_builder;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+      $container->get('form_builder')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state): array {
+    $form = parent::blockForm($form, $form_state);
+    $config = $this->getConfiguration();
+
+    $form['form_action'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Form action'),
+      '#description' => $this->t('The url the form should submit to.'),
+      '#default_value' => $config['form']['action'],
+      '#required' => TRUE,
+    ];
+    $form['input_name'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Input name'),
+      '#description' => $this->t('A name for the search input.'),
+      '#default_value' => $config['input']['name'],
+      '#required' => TRUE,
+    ];
+    $form['input_label'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Input Label'),
+      '#description' => $this->t('A label text for the search input.'),
+      '#default_value' => $config['input']['label'],
+    ];
+    $form['input_placeholder'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Input placeholder text'),
+      '#default_value' => $config['input']['placeholder'],
+    ];
+    $form['button_label'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Button label'),
+      '#description' => $this->t('Label text that should appear inside the button.'),
+      '#default_value' => $config['button']['label'],
+    ];
+    $form['button_type'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Button type'),
+      '#description' => $this->t('Ex: submit, button.'),
+      '#default_value' => $config['button']['type'],
+    ];
+    $form['button_icon_name'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Button icon name'),
+      '#description' => $this->t('Ex: search.'),
+      '#default_value' => $config['button']['icon']['name'],
+    ];
+    $form['button_icon_position'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Button icon position'),
+      '#description' => $this->t('The position of the icon inside the button.'),
+      '#default_value' => $config['button']['icon']['position'],
+    ];
+    $form['enable_autocomplete'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Enable autocomplete'),
+      '#default_value' => $config['view_options']['enable_autocomplete'] ?? 0,
+    ];
+    $form['view_id'] = [
```

Comment:
This field has to include a better message. If view used at oe_whitelabel (on oe_showcase is working smooth) there's an error breaking drupal: InvalidArgumentException: Search entity with ID "serapi" not found. in Drupal\search_api_autocomplete\Element\SearchApiAutocomplete::processSearchApiAutocomplete() (line 62 of /var/www/html/build/modules/contrib/search_api_autocomplete/src/Element/SearchApiAutocomplete.php).

For me there must be a different color for this widget so when configuring the site builder will take more care in filling ok.

Reply:
A validation must be added at a validateForm function to check view_id and view_display exist...if view is not avail, then don't show block, show a message on the block and log. But don't break

Reply:
add rather a check if views and search api are enabled